### PR TITLE
Fix manifest `Extra` field marshaling

### LINKED ISSUE #2021: Stateroot mismatch at testnet's 180464
There is one transaction (0xc88cc78efb1854978d24ddbbfe8dea731141c39e9c735b484cf101ece26203fe) there that works without any issues, but:
```
2021-06-20T17:26:02.356Z        ERROR   can't add SV-signed state root  {"error": "stateroot mismatch at block 180464: abf6b8a1589bda54423a241925df83a157c437a9293d706327c356fca25680b6 vs 96a878d57b98b8eae785d0db987f12727ae1860f402fbeb30237e53551d7988f"}
```

### PR DESCRIPTION
**Pull request by fyrchik**
Close #2021. 

State root at that height matches the one specified in task.

Also found and fixed another bug which affected `getapplicationlog` response.

### CODE REVIEW

File: pkg/smartcontract/manifest/manifest.go [hunk 1]
```diff
@@ -142,7 +144,24 @@ func (m *Manifest) ToStackItem() (stackitem.Item, error) {
...
+		var err error
+		if !bytes.Equal([]byte("null"), m.Extra) {
+			// The result is put directly in the database and affects state-root calculation,
+			// thus use ordered map to stay compatible with C# implementation.
+			var mp ojson.OrderedObject
+			err = ojson.Unmarshal(m.Extra, &mp)
+			if err == nil {
+				var res []byte
+				res, err = ojson.Marshal(mp)
+				if err == nil { // Should always be nil, but who knows.
```
Comment (AnnaShaleva) [inline pkg/smartcontract/manifest/manifest.go:156]: Then maybe return an error if so?

File: pkg/vm/stackitem/serialization.go [hunk 2]
```diff
@@ -129,7 +129,10 @@ func decodeBinaryStackItem(r *io.BinReader, allowInvalid bool) Item {
 	switch t {
 	case ByteArrayT, BufferT:
 		data := r.ReadVarBytes(MaxSize)
-		return NewByteArray(data)
+		if t == ByteArrayT {
+			return NewByteArray(data)
```
Comment (AnnaShaleva) [inline pkg/vm/stackitem/serialization.go:133]: `TestNEP11_OwnerOf_BalanceOf_Transfer` should also be fixed.

**Review by AnnaShaleva — CHANGES_REQUESTED** [review]

File: pkg/smartcontract/manifest/manifest.go (hunk 1 above)
Comment (fyrchik) [reply pkg/smartcontract/manifest/manifest.go:156]: `json.RawMessage` should ensure validity automatically. But `Extra` field can in theory be an array.

Comment (fyrchik) [reply pkg/smartcontract/manifest/manifest.go:156]: Well we must support this case as well, will check the behaviour of C# node.

**Review by roman-khimov — APPROVED** [review]

**Review by AnnaShaleva — APPROVED** [review]

**Pull request merged**